scripts/shellcheck.bash

```diff
@@ -22,7 +22,7 @@ shellcheck --shell bash --external-sources \
   test/fixtures/dummy_legacy_plugin/bin/* \
   test/fixtures/dummy_plugin/bin/*
 
-# check .bats files
-# TODO(jthegedus): unlock this check later
-# TODO  shellcheck --shell bats --external-sources \
-# TODO  test/*.bats
```

Comment:
The point of #1396 & #1397 was that the absence of the `--shell` flag gave a false sense of security. People thought files were being tested that were not.

I would like to keep `--shell bats`. The `--shell bash` files in the `test/` dir are covered in the above Shellcheck execution with that shell flag set.

Why exclude `SC2164`?

Why change severity?

Reply:
I'm not sure I see your understand your first point. I read on the shellcheck man page that:

> The default [if --shell is not specified] is to deduce the shell from the file's shell directive, shebang, or .bash/.bats/.dash/.ksh extension, in that order.  sh refers to POSIX sh (not the system's), and will warn of portability issues.

Since the files extensions of each test in `test/*` match how it should be linted, I'm not sure what the "false sense of security is" - the way I understood it, the only issue are files that end with `.sh` - since that is potentially ambiguous semantically (without a shebang or `#shellcheck shell=bash` entry at the top). In my opinion, `shellcheck.bash` doesn't make all that much sense to me - if we used the command `shellcheck --external-sources ./**/*.{sh,bash,bats}` from the start, then we wouldn't be in the predicament of the `test/` directory not being linted in the first place. With how we currently have it, the checking against the type of shell can potentially differ from the filename extension, and we can potentially skip files if the directory structure changes and we forget to update the command. I was going to make a PR for this later, but I guess it's good we can discuss this now.

Reply:
I disabled `SC2164` because it was not relevant since Bats automatically applies `-e` mode, and ShellCheck doesn't know that - I'll add a comment for each rule I ignore.

The severity was changed temporarily, just to show that "all warnings pass" - the [next pr](#1453) properly removes it, so everything is tested

Reply:
@hyperupcall the script is the way it is for historical reasons. I would like to keep the definition of files and which shell we test against explicit for now, 1 for execution of shellcheck for each type of `--shell`. We can revisit later once we decide on `sh` vs `bash` as outlined in #1396

I would prefer not to ignore SC rules across the whole execution from the command, but rather use comments and ignore per-occurrence and then per-file.

Reply:
With the check at the top of each file we can in future use the single-line shellcheck you proposed above without the exclude leaking into other files. Thanks